kernel/src/log_compaction/tests.rs

```diff
@@ -0,0 +1,845 @@
+use super::writer::{should_compact, LogCompactionDataIterator, LogCompactionWriter};
+use super::COMPACTION_ACTIONS_SCHEMA;
+use crate::path::ParsedLogPath;
+use crate::{DeltaResult, EngineData, FileMeta, Version};
+use std::sync::Arc;
+use url::Url;
+
+#[test]
+fn test_log_compaction_writer_creation() {
+    let table_root = Url::parse("memory:///test-table").unwrap();
```

Comment:
I initially had file-based root at /tmp. Caused Windows CI failures. Switched to memory:  faster, portable, and consistent with existing kernel tests.